tests/runtime/op_reg/kernel_aot_test.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import logging
+import unittest
+
+import torch
+import torch.nn as nn
+
+import shark_turbine.aot as aot
+import shark_turbine.ops as ops
+
+from shark_turbine.transforms.general.custom_op_expansion import ExpandCustomOpsPass
+
+
+class MLP(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.layer0 = nn.Linear(8, 8, bias=True)
+        self.layer1 = nn.Linear(8, 4, bias=True)
+        self.layer2 = nn.Linear(4, 2, bias=True)
+        self.layer3 = nn.Linear(2, 2, bias=True)
+
+    def forward(self, x: torch.Tensor):
+        x = self.layer0(x)
+        x = torch.sigmoid(x)
+        ops.iree.trace_tensor("LAYER0", x)
+        x = self.layer1(x)
+        x = torch.sigmoid(x)
+        ops.iree.trace_tensor("LAYER1", x)
+        x = self.layer2(x)
+        x = torch.sigmoid(x)
```

Comment:
This is cool! would we be using these instead of compiledModule in the future?

Reply:
Different use cases. But this unifies things so that you can tie arbitrary ir shenanigans to pytorch ops and have them work the same whether in eager dispatch, torch.compile, or aot/export paths